src/main/java/commands/GoModuleCommand.java

```diff
@@ -32,32 +40,29 @@ public void execute(Ui ui, Access access, Storage storage) throws IOException {
     }
 
     private String goModule(Access access, Storage storage) throws IOException {
-        boolean isLevelExist = false;
+        assert access.isAdminLevel() : "Not admin level";
         String result = "";
-        String currentModule = this.module;
-        ModuleList modules = access.getAdmin().getModules();
-        ArrayList<Module> allModules = modules.getAllModules();
-        for (Module module : allModules) {
-            String moduleName = module.getModuleName();
-            if (currentModule.equalsIgnoreCase(moduleName)) {
-                access.setModuleLevel(moduleName);
-                isLevelExist = true;
-                try {
-                    ArrayList<Chapter> chapters = storage.loadChapter(module.getModuleName());
-                    if (chapters.size() == 0) {
-                        result = "This is a new module, you can try to add chapters inside!";
-                    }
-                    module.setChapters(chapters);
-                    access.setModule(module);
-                } catch (FileNotFoundException e) {
-                    result = "The module folder cannot be found.";
-                }
-                break;
+        try {
+            ModuleList modules = access.getAdmin().getModules();
+            ArrayList<Module> allModules = modules.getAllModules();
+            Module module = allModules.get(moduleIndex);
+            logger.info("Going into module: " + module.toString());
+            access.setModuleLevel(module.getModuleName());
```

Comment:
this line can be shift to after line 56: access.setModule(module)
so you load the chapters first. and if the module cannot be found then access wont set anything yet
so access.setModuleLevel should be after loading the chapters